src/NumSharp.Core/Operations/Elementwise/NDArray.Greater.cs

```diff
@@ -1,7 +1,358 @@
-﻿namespace NumSharp
+﻿using System;
+using NumSharp.Backends;
+using NumSharp.Generic;
+using NumSharp.Utilities;
+
+namespace NumSharp
 {
     public partial class NDArray
     {
+        public static unsafe NumSharp.Generic.NDArray<bool> operator >(NDArray lhs, NDArray rhs)
+        {
+#if _REGEN
+            if(lhs.typecode != rhs.typecode)
+            {
+                throw new NotImplementedException("please make sure operands have the same data type");
```

Comment:
They do not have to be of the same dtype.
Like all operators, this has to support all dtypes combinations.